kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/annotation/specAnnotations.kt

```diff
@@ -16,3 +19,17 @@ annotation class Tags(vararg val values: String)
 @Target(AnnotationTarget.CLASS)
 @Retention(AnnotationRetention.RUNTIME)
 annotation class Ignored
+
+/**
+ * Attach to [io.kotest.core.spec.Spec], and the logic inside [enabledIf] will be executed
+ * to define if a test will be instantiated or executed.
+ * 
+ * [SpecEnabaledIf] must contain a no-arg constructor as it will be used via reflection.
+ */
+@Target(AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class EnabledIf<T : SpecEnabledIf>(val enabledIf: KClass<T>)
```

Comment:
I was thinking something along the lines of

`annotation class EnabledIf(val test: (KClass<*> -> Boolean)`